program_intent_engagement/apps/api/serializers.py

```diff
@@ -1,4 +1,33 @@
-# Serializers that can be shared across multiple versions of the API
-# should be created here. As the API evolves, serializers may become more
-# specific to a particular version of the API. In this case, the serializers
-# in question should be moved to versioned sub-package.
+"""
+Serializers for the program intent engagement API
+
+These serializers are shared across multiple API versions.
+Version-specific serializers are located in the versioned subpackage.
+"""
+from rest_framework import serializers
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+
+
+class ProgramIntentSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the Program Intent Model
+    """
+
+    # TODO: should all fields be required?
+    # TODO: is foreign key not included here?
```

Comment:
The incoming thing is a new program intent, so it can't know anything about the database. The lms user ID will be in the JWT token coming with the request.

Reply:
the demographics API has to solve similar problems where demographics can only be changed for the user who is changing them
https://github.com/edx/demographics/blob/master/demographics/apps/user_demographics/rest_api/v1/permissions.py is how it picks up user ID and checks permission
https://github.com/edx/demographics/blob/4d948827e7b9e4b07045227e87e1751536a0e748/demographics/apps/user_demographics/rest_api/v1/views.py#L38 we'll need this JWT authentication class on the view so that it does all the JWT magic (I don't think we need the session class)
this is the change that makes is so when we do JWT authentication and create a local user with it, that LMS global user ID is saved into lms_user_id locally https://github.com/edx/program-intent-engagement/commit/4de0f0db375fd5ce47ea568a0c7c3157193c8ccf

Reply:
this is stuff @m-pitera 's endpoint needs as well

Reply:
to clarify my first link I don't think we need to check permission, it's more that we need to look at the user and say this action is for this user - so any insert is that user, and any query is across that user